### ISSUE TITLE
InvalidCastException on ImagesFromAsync

### ISSUE THREAD

**Issue by Matti-Koopa** [DrJohnMelville/Pdf#39]
Code is:
`await document.ImagesFromAsync(page)`
Pdf:
(removed)

Exception:
```
System.InvalidCastException: Unable to cast object of type 'Melville.Pdf.ImageExtractor.ImageExtractorTarget' to type 'Melville.Pdf.TextExtractor.ExtractTextRender'.
   at Melville.Pdf.TextExtractor.ExtractingFont.BeginFontWrite(IFontTarget target)
   at Melville.Pdf.Model.Renderers.PdfStringWriter..ctor(IFontTarget target)
   at Melville.Pdf.Model.Renderers.RenderEngine.GetSpacedStringBuilder()
   at Melville.Pdf.Model.Renderers.RenderEngine.ShowStringAsync(ReadOnlyMemory`1 decodedString)
   at Melville.Postscript.Interpreter.InterpreterState.PostscriptEngine.ExecuteTokenAsync(PostscriptValue token)
   at Melville.Postscript.Interpreter.InterpreterState.PostscriptEngine.ExecuteTokenAsync(PostscriptValue token)
   at Melville.Postscript.Interpreter.InterpreterState.PostscriptEngine.RunSingleStepAsync(PostscriptValue token)
   at Melville.Postscript.Interpreter.InterpreterState.PostscriptEngine.MainExecutionLoopAsync()
   at Melville.Postscript.Interpreter.InterpreterState.PostscriptEngine.ExecuteAsync(ITokenSource tokens)
   at Melville.Pdf.LowLevel.Parsing.ContentStreams.ContentStreamParser.ParseAsync(IByteSource source)
   at Melville.Pdf.Model.Renderers.RenderEngine.RunContentStreamAsync()
   at Melville.Pdf.Model.Renderers.DocumentRenderers.DocumentRenderer.RenderPageToAsync(Int32 oneBasedPageNumber, Func`3 target)
   at Melville.Pdf.ImageExtractor.ImageExtractorFacade.ImagesFromAsync(DocumentRenderer dr, Int32 page)
```

**Comment by DrJohnMelville** [comment]
OK, I can repro this bug.  Here is the repro code:

````
public class Bugs
{
    public async Task ReadAllImages()
    {
        var bytes =
            File.ReadAllBytes(
                @"C:\Users\jom252\OneDrive - Medical University of South Carolina\Documents\Scratch\Pdf\BUGS\1483757-2022-10.pdf");
        var doc = await new PdfReader().ReadFromAsync(bytes);
        var text = await doc.PageTextAsync(1); // if you comment this out everything works
        var ret = doc.ImagesFromAsync();
        await foreach (var image in ret)
        {
            GC.KeepAlive(image);
        }
    }
}
````

What is going on here is that when you extract the text, the documentrenderer is caching a font, and it is caching a ExtractingFont.  ExtractingFont expects to render to an ExtractTextRenderTarget, which the image extractor does not render to.  The solution is to include the type of the renderer in the cached font result.

**Comment by DrJohnMelville** [comment]
Fixed in build f832b21c
